app/src/main/java/com/khmelenko/lab/varis/log/TravisCommands.java

```diff
@@ -0,0 +1,83 @@
+package com.khmelenko.lab.varis.log;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.reactivex.Observable;
+
+public class TravisCommands {
+
+    private static final String TRAVIS_TIME = "travis_time";
+    private static final String TRAVIS_FOLD = "travis_fold";
+    private static final String START = "start";
+    private static final String END = "end";
+    private static final Pattern TRAVIS_COMMAND = Pattern.compile('(' + TRAVIS_FOLD + '|' + TRAVIS_TIME + "):(" + START + '|' + END + "):(\\S+)");
+
+    /**
+     * Takes a list of TextLeafs and splits them when it contains a travis command into a
+     * TextLeaf + TravisCommandLeaf + TextLeaf.
+     */
+    public static List<LogEntryComponent> preProcessTextLeafs(List<TextLeaf> logEntryComponents) {
+        return Observable.fromIterable(logEntryComponents)
+                .flatMap(logEntryComponent -> Observable.fromIterable(splitAtTravisCommands(logEntryComponent)))
+                .toList()
+                .blockingGet();
```

Comment:
`blockingSingle()` should be better in this case, because there is only 1 item in Single and it's a List

Reply:
Not sure if I get what you mean. It's on purpose to get a List here...

Reply:
ah, yeah, sorry, my mistake, you are right here